src/utils/ManifestLoader.ts

```diff
@@ -107,3 +108,29 @@ try {
 }
 
 export { nodeSection };
+
+export function isLeaf(obj: any): obj is Leaf {
```

Comment:
Not a big fan of `any`, since it requires you to disable ESLint. Since it only makes sense to call these functions with tree nodes, I think you could define a `type TreeNode = RootNode | RootChild | ParentNode | Leaf` then change the type signature to that. For example in `isLeaf`, it would just become
```ts
export function isLeaf(obj: TreeNode): obj is Leaf {
  return "name" in obj && "key" in obj && "type" in obj && !obj.children;
}
```